File: Leetcode/2114_maximum_word_in_sentence/solution.py
"""
  link: https://leetcode.com/problems/maximum-number-of-words-found-in-sentences/
  tag: Array, String
"""


def maximumWordInSentence(sentences):
    maxWord = 0

    for sentence in sentences:
        maxWord = max(maxWord, len(sentence.split()))

    return maxWord


print(maximumWordInSentence(["alice and bob love leetcode",
      "i think so too", "this is great thanks very much"]))

File: Leetcode/2114_maximum_word_in_sentence/solution.go
package main

import (
	"fmt"
	"strings"
)

func maximumWordInSentence(sentences []string) int {
	count := 0;

	for _, sentence := range sentences {
		wordCount :=  len(strings.Split(sentence, " "))

		if wordCount > count {
			count = wordCount;
		}
	}

	return count;
}

func main() {
	sentences := []string {"alice and bob love leetcode", "i think so too", "this is great thanks very much"}

	fmt.Println(maximumWordInSentence((sentences)))
}